src/buildstream_plugins/sources/docker.py

```diff
@@ -387,22 +387,23 @@ def configure(self, node):
             self.original_registry_url = node.get_str("registry-url", _DOCKER_HUB_URL)
             self.registry_url = self.translate_url(self.original_registry_url)
 
-        if "ref" in node:
-            self.digest = self._ref_to_digest(node.get_str("ref"))
-        else:
-            self.digest = None
         self.tag = node.get_str("track", "") or None
 
         self.architecture = node.get_str("architecture", "") or default_architecture()
         self.os = node.get_str("os", "") or default_os()
 
-        if not (self.digest or self.tag):
-            raise SourceError("{}: Must specify either 'ref' or 'track' parameters".format(self))
```

Comment:
This check seems superfluous to me. Whenever self.digest is None, is_resolved return False.

Reply:
I think the point is the same as the git plugin: we should either provide the ref or enough configuration to track. This is supposed to give an early error to the user that their configuration can't work.

This might no longer be relevant, I'm just talking about the intent.

Reply:
Good point. I haven't seen this pattern in any other plugin but indeed git plugin from this repo uses same pattern. It seems to me main difference with this removed is that when track is None, plugin will skip tracking even if there is no ref.

Reply:
Plugin will not try to fetch anything because as said is_resolved returns False.

Reply:
The UX is slightly changed with this, maybe slightly regressed (the user will have to figure out on their own, why their docker source which did not specify any ref or any tag, didn't pull anything and cannot be built) but this seems reasonable.

I want to include this in the next release so merging as is.